test/AccTabSizeChanged.js

```diff
@@ -0,0 +1,172 @@
+const Operations = artifacts.require("Operations");
+const utils = require("../utils")
+
+const scriptName = utils.scriptName(__filename);
+const testData = utils.testData;
+
+const { uint32_id, string_name, uint32_price } = testData.device;
+
+const bytes_type_user = utils.str2bytes("user");
+const bytes_type_owner= utils.str2bytes("owner");
+const uint256_msg = testData.uint256_msg;
+const right = true;
+
+contract('Operations', (accounts) => {
+    let title = `gas used by [${scriptName}]`
+
+    var owner;
+    var user;
+    before(`init accounts in current network. [${scriptName}]`, async () => {
+        owner = await utils.createAccount(web3, utils.prikeys[testData.ownerIndex]);
+        user = await utils.createAccount(web3, utils.prikeys[testData.userIndex]);
+    });
+
+    const addFakedatas = async (contractInstance, num, fakedata) => {
+        for (var i = 0; i < num; i++) {
+            await contractInstance.addAccTab(
+                fakedata.id,
+                fakedata.from,
+                fakedata.to,
+                fakedata.expire,
+                fakedata.right,
+                {
+                    from: accounts[1]
+                }
+            );
+        }
+    }
+
+    const addFakedatas_ = async (contractInstance, num, fakedata) => {
+        const batchSize = 60;
+
+        while (num > 0) {
+
+            var tasks = [];
+            var taskNum = num > batchSize ? batchSize : num;
+
+            for (var i = 0; i < taskNum; i++) {
+                tasks.push(contractInstance.addAccTab(
+                    fakedata.id,
+                    fakedata.from,
+                    fakedata.to,
+                    fakedata.expire,
+                    fakedata.right,
+                    {
+                        from: accounts[1]
+                    }
+                ));
+            }
+
+            await Promise.all(tasks).then((results) => {
+                console.log(results.length);
+            });
+
+            num -= batchSize;
+            console.log(`add ${taskNum} fake data`)
+        }
+    };
```

Comment:
It looks like `addFakedatas_` is no longer needed.

Reply:
yes, It is no longer needed.
I thought that running multiple async functions at the same time would speed up the insert operation.
but I got errors below :

```
ProviderError: 
Could not connect to your Ethereum client.
Please check that your Ethereum client:
    - is running
    - is accepting RPC connections (i.e., "--rpc" option is used in geth)
    - is accessible over the network
    - is properly configured in your Truffle configuration file (truffle-config.js) 
```

May be ? truffle's test network doesn't support so many concurrency ?